core/src/main/java/org/apache/iceberg/SystemProperties.java

```diff
@@ -39,11 +39,24 @@ private SystemProperties() {
    */
   public static final String SCAN_THREAD_POOL_ENABLED = "iceberg.scan.plan-in-worker-pool";
 
-  static boolean getBoolean(String systemProperty, boolean defaultValue) {
+  /**
+   * Whether to use the shared worker pool when planning table scans.
+   */
+  public static final String READ_DELETE_FILES_WORKER_POOL_SIZE = "iceberg.worker.read-deletes-num-threads";
+
+  public static boolean getBoolean(String systemProperty, boolean defaultValue) {
```

Comment:
As I said elsewhere, I don't think that this feature should be controlled through a system property. This should be a Flin-specific property for now and we can introduce a similar config for Spark later. But since this violates Spark's threading model on executors, we don't want to make this global.